### ISSUE TITLE
`setMaterialPriority` not working for main material grid view

### ISSUE THREAD

**Issue by Krutoy242** [Project-RT/RandomTweaker#67]
Im using this methof to rearrange materials in book:

```less
for matName, power in scripts.equipment.equipData.defaultWeaponMats {
  mods.randomtweaker.tconstruct.IBook.setMaterialPriority(matName, power);
}
```

Comes out that only detailed pages arranged (were we can see stats and applicable tool parts), but main "grid" view is stay the same:

![image](https://github.com/Project-RT/RandomTweaker/assets/936618/c52d401c-3355-4a4f-8581-a1101b071e06)

[debug.log](https://github.com/Project-RT/RandomTweaker/files/13226985/debug.log)

**Comment by ikexing-cn** [comment]
To be honest, I'm not entirely sure what is happening from your description. To avoid ambiguity, it needs to be made clear that the implementation of the `setMaterialPriority` method will only handle with the material that's set, while the rest will retain the default ordering.

**Comment by Krutoy242** [comment]
I thoung, setMaterialPriority('paper', -1000)` will place paper here:

![image](https://github.com/Project-RT/RandomTweaker/assets/936618/9e022f8b-50a3-4db3-8806-d1a829eaaa11)

**Comment by ikexing-cn** [comment]
try `setMaterialPriority('paper', 1000)``

**Comment by Krutoy242** [comment]
Not working too.

**Comment by Krutoy242** [comment]
There, i printed messages about values to be sure:

This is my ZS code:
```ts
for matName, power in scripts.equipment.equipData.defaultWeaponMats {
  print('randomtweaker: setting '~matName~' priority: '~(-power));
  mods.randomtweaker.tconstruct.IBook.setMaterialPriority(matName, -power);
}
```

This is cafttweaker.log output:
```less
[INITIALIZATION][CLIENT][INFO] randomtweaker: setting prismarine priority: -80
[INITIALIZATION][CLIENT][INFO] Setting material prismarine priority to -80
...
[INITIALIZATION][CLIENT][INFO] randomtweaker: setting paper priority: -21
[INITIALIZATION][CLIENT][INFO] Setting material paper priority to -21
...
[INITIALIZATION][CLIENT][INFO] randomtweaker: setting endstone priority: -77
[INITIALIZATION][CLIENT][INFO] Setting material endstone priority to -77
```

But we still seeing them near each other:

![image](https://github.com/Project-RT/RandomTweaker/assets/936618/333ed459-62be-46f8-90f1-49ea367cffd9)

**Comment by ikexing-cn** [comment]
In fact, this is an expected result.
Simply setting a priority for some materials will only result in them being inserted in an inconsistent order. 
I think your requirements need to be sorted for all materials to control the specific insertion order.

**Comment by Krutoy242** [comment]
Ok, seems like I just come up with wrong example. I will make another example to show that order actually not changing.

**Comment by Krutoy242** [comment]
There is another example showing that sorting does not work.

Crafttweaker.log output:
```less
Setting material paper priority to 21
Setting material stone priority to 31
Setting material construction_alloy priority to 41
Setting material flint priority to 49
```

What i see:
![image](https://github.com/Project-RT/RandomTweaker/assets/936618/9be7c885-038b-4b73-b9dd-6546502bf962)

**Comment by ikexing-cn** [comment]
Fixed, [#1397ec6](https://github.com/Project-RT/RandomTweaker/commit/1397ec6a4f67ebbe8ac802d0a026df3a4dc04628)
